### PR TITLE
Extend Cols API

### PR DESCRIPTION
**Pull request by bkamins**
Following https://github.com/JuliaData/DataFrames.jl/pull/2881/files#r711943431.

### CODE REVIEW

File: src/DataAPI.jl [hunk 1]
```diff
@@ -161,8 +161,13 @@ end
 
 """
     Cols(cols...)
+    Cols(f::Function)
 
 Select the union of the selections in `cols`. If `cols == ()`, select no columns.
+
+If the only positional argument is a `Function` `f` then select the columns whose
+names passed to the `f` predicate as strings return `true`.
+As a special case if `:` is passed (the `Colon` function) select all columns.
```
Comment (nalimilan) [inline src/DataAPI.jl:170]: Do we really need to mention `:`? It sounds quite obvious that if `:` is allowed as a selector to get all columns, passing it selects all columns.
```suggestion
```

Comment (bkamins) [reply src/DataAPI.jl:170]: Technically `:` is a `Function` so this is a special case. Do you still think it should be removed?

Comment (nalimilan) [reply src/DataAPI.jl:170]: Ah, got it! I wouldn't mention it though, as nobody would expect `:` to be treated as a predicate on strings (anyway that would error).

Comment (bkamins) [reply src/DataAPI.jl:170]: OK - removing.

**Review by nalimilan — APPROVED** [review]

**Comment by bkamins** [comment]
Thank you!

**Pull request merged**